client/me/account-password/index.jsx

```diff
@@ -38,24 +39,30 @@ const AccountPassword = createReactClass( {
 
 	mixins: [ observe( 'accountPasswordData' ) ],
 
-	componentDidMount: function () {
+	componentDidMount() {
 		this.debouncedPasswordValidate = debounce( this.validatePassword, 300 );
 	},
 
-	componentWillUnmount: function () {
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.isPendingPasswordChange && ! this.props.isPendingPasswordChange ) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState( { isUnsaved: false } );
```

Comment:
Not an ideal solution in the long run, but will do the job for now to avoid a larger refactor.